apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/parser/graphql/ast/schema_validation.kt

```diff
@@ -0,0 +1,233 @@
+package com.apollographql.apollo.compiler.parser.graphql.ast
+
+import com.apollographql.apollo.compiler.parser.error.ParseException
+
+fun GQLDocument.validateAsSchema(): GQLDocument {
+  validateNotExecutable()
+  validateUniqueSchemaDefinition()
+  validateTypeNames()
+  validateDirectiveNames()
+  validateInterfaces()
+  validateObjects()
+  return this
+}
+
+fun GQLDocument.mergeTypeExtensions(): GQLDocument {
```

Comment:
is it a right place for this function? Looks like all others related to validation this one is merging definitions